app/validators/age_validator.rb

```diff
@@ -0,0 +1,39 @@
+class AgeValidator < ActiveModel::EachValidator
+  MAX_AGE = 120
+
+  def validate_each(record, attribute, value)
+    if value.present?
+      check_is_date(record, attribute, value) &&
+        check_range(record, attribute, value)
+    end
+  end
+
+private
+
+  def check_is_date(record, attribute, value)
+    if value.is_a?(Date)
+      true
+    else
+      record.errors.add(
+        attribute,
+        I18n.t('age_validator.errors.invalid_date')
```

Comment:
There might be no need to use `I18n`, if you just substituted it for `invalid_date` the error message I think will still be translated, at least that's what happens with custom validations in models.